fix(ssi): test all webhooks in integration test

betterengineering opened the pull request:
### What does this PR do?
This commit fixes an issue where we only tested the auto instrumentation webhook for SSI in isolation when in practice we use it in combination with the other webhooks. In doing so, I found a bug where we would still mutate a disabled namespace for the tags from labels or config webhooks when the customer had:
* A disabled namespace defined
* A pod with local SDK injection existed in the disabled namespace

### Motivation
When we create the auto instrumentation webhook, we use all three mutators:
https://github.com/DataDog/datadog-agent/blob/0017ef02f9a5e677d383026b0382034e344c9ec4/pkg/clusteragent/admission/controllers/webhook/controller_base.go#L197-L204

However, in our integration tests, we only tested the auto instrumentation mutator in isolation. This change makes sure our tests run against the three mutators as a unit.

### Describe how you validated your changes
The additional unit tests

### Additional Notes
In order to test the webhooks as a unit, I needed to ensure all workloads had a parent deployment and a namespace that exists in the workload metadata. This required me to do a big refactor to utilize the `common.FakePodSpec` that handles connecting the associated deployment. Honestly, I should have done this from the beginning as this is a much cleaner implementation.

betterengineering committed:
fix(ssi): test all webhooks in integration test

This commit fixes an issue where we only tested the auto instrumentation
webhook for SSI in isolation when in practice we use it in combination
with the other webhooks.

buraizu commented on releasenotes-dca/notes/fix-ssi-webhook-disabled-namespaces-035fb4bb6ce6c521.yaml:
```diff
@@ -0,0 +1,13 @@
+---
+fixes:
+  - |
+    Customers using Single Step Instrumentation with both configuration based targeting (enabled namespace or
```
```suggestion
    Customers using Single Step Instrumentation with both configuration-based targeting (enabled namespace or
```

domalessi reviewed (approved):
Suggested a tweak to the release note for clarity, but approving so you're not blocked on me!

betterengineering committed:
Update releasenotes-dca/notes/fix-ssi-webhook-disabled-namespaces-035fb4bb6ce6c521.yaml

Co-authored-by: domalessi <111786334+domalessi@users.noreply.github.com>

betterengineering committed:
Merge branch 'main' into mark.spicer/run-all-webhooks-for-test

betterengineering committed:
fix(ssi): resolve bug with disabled label in enabled namespace (#43543)

### What does this PR do?
This commit resolves a bug where a pod with a disabled label in an
enabled namespace would still receive injection.

### Motivation
When working through a support case, I noticed that we did not respect
the disabled label in an enabled namespace.

### Describe how you validated your changes
The additional unit tests.

### Additional Notes
This entire webhook is scheduled to be refactored. When doing so, we
should consider a more sophisticated version of the `annotationResult`:
<img width="1062" height="481" alt="Screenshot 2025-11-26 at 11 56
36 AM"
src="https://github.com/user-attachments/assets/94df3a41-0bf6-481b-9096-4ef1c2e5ee87"
/>

---------

Co-authored-by: domalessi <111786334+domalessi@users.noreply.github.com>

betterengineering committed:
Merge branch 'main' into mark.spicer/run-all-webhooks-for-test

betterengineering commented:
/merge
